src/cryojax/simulator/_detector.py

```diff
@@ -3,117 +3,161 @@
+            # and return.
+            expected_electron_events = irfftn(
+                fourier_expected_electron_events, s=config.padded_shape
+            )
+            return rfftn(self.sample(key, expected_electron_events))
 
 
-class NullDetector(AbstractDetector, strict=True):
-    """
-    A 'null' detector.
-    """
+class NullDetector(AbstractDetector):
+    """A null detector model."""
 
-    is_sample_real: ClassVar[bool] = False
+    @override
+    def __init__(self):
+        self.dqe = NullDQE()
 
     @override
     def sample(
+        self, key: PRNGKeyArray, expected_electron_events: RealImage
+    ) -> RealImage:
+        return expected_electron_events
+
+    @override
+    def __call__(
         self,
-        key: PRNGKeyArray,
-        image: ComplexImage,
-        coords_or_freqs: ImageCoords,
+        fourier_squared_wavefunction_at_detector_plane: ComplexImage,
+        dose: ElectronDose,
+        config: ImageConfig,
+        key: Optional[PRNGKeyArray] = None,
     ) -> ComplexImage:
-        return image
+        return fourier_squared_wavefunction_at_detector_plane
 
 
 class GaussianDetector(AbstractDetector, strict=True):
+    """A detector with a gaussian noise model. This is the gaussian limit
+    of `PoissonDetector`.
     """
-    A detector with a gaussian noise model. By default,
-    this is a white noise model.
-
-    Attributes
-    ----------
-    variance :
-        A kernel that computes the variance
-        of the detector noise. By default,
-        ``Constant()``.
-    """
-
-    is_sample_real: ClassVar[bool] = False
-
-    variance: FourierOperatorLike = field(default_factory=Constant)
 
     @override
     def sample(
-        self,
-        key: PRNGKeyArray,
-        image: ComplexImage,
-        coords_or_freqs: ImageCoords,
-    ) -> ComplexImage:
-        noise = self.variance(coords_or_freqs) * jr.normal(
-            key, shape=coords_or_freqs.shape[0:-1], dtype=complex
-        )
-        noise = noise.at[0, 0].set(0.0 + 0.0j)
-        return image + noise
+        self, key: PRNGKeyArray, expected_electron_events: RealImage
+    ) -> RealImage:
+        return expected_electron_events + jnp.sqrt(
+            expected_electron_events
+        ) * jr.normal(key, expected_electron_events.shape)
 
 
 class PoissonDetector(AbstractDetector, strict=True):
-    """
-    A detector with a poisson noise model.
-
-    NOTE: This is untested and very much in a beta version.
-    """
-
-    is_sample_real: ClassVar[bool] = True
+    """A detector with a poisson noise model."""
 
     @override
     def sample(
-        self,
-        key: PRNGKeyArray,
-        image: RealImage,
-        coords_or_freqs: ImageCoords,
+        self, key: PRNGKeyArray, expected_electron_events: RealImage
     ) -> RealImage:
-        return jr.poisson(key, image).astype(float)
+        return jr.poisson(key, expected_electron_events).astype(float)
```

Comment:
you could illustrate a case of having `expected_electron_events` as `
dose*image + background` in a notebook.

Doing inference in this model could have dose and background as a global scalar parameters, and image as a per measurement parameter  of shape (n_pix,n_pix).

Reply:
In this case, what is background? Is this to account for the possibility of not having the right global mean?